### PR TITLE
SC-1740 - Search Service dockerization

### PR DESCRIPTION
**Pull request by PradyumnaNagendra**

### CODE REVIEW

File: ansible/inventory/env/group_vars/all.yml [hunk 1]
```diff
@@ -696,3 +696,7 @@ sunbird_portal_updateLoginTimeEnabled: false
 #sunbird_offline_azure_storage_account: "" #added this var for adopter usecase
 offline_installer_container_name: "" #added this var for adopter usecase
 cloud_storage_url: "{{ sunbird_public_storage_account_name }}.blob.core.windows.net"
+
+# Search-service
+search_index_host: "{{ groups['composite-search-cluster'][0] }}:9200"
```
Comment (SMYALTAMASH) [inline ansible/inventory/env/group_vars/all.yml]: What if first host is unreachable in cluster? Shouldn't it take entire cluster comma separated IP.

Comment (PradyumnaNagendra) [reply ansible/inventory/env/group_vars/all.yml]: done

**Comment by SMYALTAMASH** [comment]
@keshavprasadms please review this

File: ansible/roles/stack-sunbird/tasks/search-service.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  become: yes
+  shell: "docker config rm search-service_logback.xml"
+  ignore_errors: yes
+  
+- name: Copy new logback.xml
+  become: yes
+  shell: "docker config create search-service_logback.xml /home/deployer/config/search-service_logback.xml"
+
+- name: Deploy search-service
+  shell: "docker service create --with-registry-auth --replicas {{ search_service_replicas }} -p 9002:9000  --name search-service --hostname search-service --reserve-memory {{ search_service_reservation_memory }} --limit-memory {{ search_service_limit_memory }} --limit-cpu {{ search_service_limit_cpu }} --reserve-cpu {{ search_service_reservation_cpu }} --health-cmd 'wget -qO- search-service:9000/health || exit 1' --health-timeout 3s --health-retries 3  --network application_default --config source=search-service.conf,target=/home/sunbird/search-service-1.0-SNAPSHOT/config/application.conf,mode=0644 --config source=search-service_logback.xml,target=/home/sunbird/search-service-1.0-SNAPSHOT/config/logback.xml,mode=0644  {{hub_org}}/{{image_name}}:{{image_tag}}"
```
Comment (keshavprasadms) [inline ansible/roles/stack-sunbird/tasks/search-service.yml]: Port should be 9004

File: ansible/roles/stack-sunbird/templates/search-service_logback.xml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        <!-- block the logging application thread if the kafka appender cannot keep up with sending the log messages -->
+        <deliveryStrategy class="com.github.danielwegener.logback.kafka.delivery.BlockingDeliveryStrategy">
+            <!-- wait indefinitely until the kafka producer was able to send the message -->
+            <timeout>0</timeout>
+        </deliveryStrategy>
+		
+		<!-- each <producerConfig> translates to regular kafka-client config (format: key=value) -->
+		<!-- producer configs are documented here: https://kafka.apache.org/documentation.html#newproducerconfigs -->
+		<!-- bootstrap.servers is the only mandatory producerConfig -->
+		<producerConfig>bootstrap.servers={{lp_kafka_url}}</producerConfig>
```
Comment (keshavprasadms) [inline ansible/roles/stack-sunbird/templates/search-service_logback.xml]: What is this variable lp_kafka_url? Please define in the inventory in the template below

Comment (PradyumnaNagendra) [reply ansible/roles/stack-sunbird/templates/search-service_logback.xml]: lp_kafka_url -  is already present as part of content-service.

File: ansible/roles/stack-sunbird/tasks/search-service.yml (hunk 2 above)
Comment (PradyumnaNagendra) [reply ansible/roles/stack-sunbird/tasks/search-service.yml]: done

**Pull request merged**